src/app/static/src/tests/components/runWorkflow/runWorkflowRun.test.ts

```diff
@@ -1,64 +1,262 @@
+            expect(mockAxios.history.get[1].url).toBe("http://app/workflows");
+            expect(wrapper.vm.$data.error).toStrictEqual("")
+            expect(wrapper.vm.$data.defaultMessage).toStrictEqual("")
+            expect(wrapper.vm.$data.workflows).toStrictEqual(workflowSummaryMetadata)
+
+            const changelog = wrapper.find("#workflow-changelog-div")
+            expect(changelog.find("label").text()).toBe("Changelog message")
+            expect(changelog.find("input").exists()).toBe(true)
+            done()
+        })
     })
 
-    it(`can render workflow-tags elements`, () => {
+    it(`can set workflow-changelog message`, async (done) => {
         const wrapper = getWrapper()
+
+        setTimeout(async () => {
+            const changelog = wrapper.find("#workflow-changelog-div")
+            expect(changelog.find("label").text()).toBe("Changelog message")
+            const input = changelog.find("input")
+            expect(changelog.find("input").exists()).toBe(true)
+
+            await input.setValue("test message input")
+            const inputValue = input.element as HTMLInputElement
+            expect(inputValue.value).toBe("test message input")
+            expect(wrapper.vm.$data.changeLogMessage).toEqual("test message input")
+            done()
+        })
+    })
+
+    it(`can render workflow-changelog-type`, async (done) => {
+        const wrapper = getWrapper()
+
+        setTimeout(async () => {
+            expect(mockAxios.history.get.length).toBe(2);
+            expect(mockAxios.history.get[0].url).toBe("http://app/report/run-metadata");
+            expect(mockAxios.history.get[1].url).toBe("http://app/workflows");
+            expect(wrapper.vm.$data.error).toStrictEqual("")
+            expect(wrapper.vm.$data.defaultMessage).toStrictEqual("")
+            expect(wrapper.vm.$data.workflows).toStrictEqual(workflowSummaryMetadata)
+
+            const changelogType = wrapper.find("#workflow-changelog-type-div")
+            expect(wrapper.find("#workflow-changelog-type-div label").text()).toEqual("Changelog type")
+            const options = changelogType.find("select").findAll("option")
+            expect(options.at(0).text()).toBe(changelogTypes[0]);
+            expect(options.at(1).text()).toBe(changelogTypes[1]);
+            done()
+        })
+    })
+
+    it(`can select workflow-changelog-type`, async (done) => {
+        const wrapper = getWrapper()
+
+        setTimeout(async () => {
+            const changelogType = wrapper.find("#workflow-changelog-type-div")
+            expect(wrapper.find("#workflow-changelog-type-div label").text()).toEqual("Changelog type")
+            const selectOptions = changelogType.find("select")
+            selectOptions.setValue(changelogTypes[1])
+            const optionsValue = selectOptions.element as HTMLSelectElement
+            expect(optionsValue.value).toEqual(changelogTypes[1])
```

Comment:
This is just checking that the value in the element has changed, should really check that the data value `selectedInstances` has changed (although this test will change anyway to test of emitted update if wizard used to manage metadata).

Reply:
I do not think we will be emitting Changelog data and if not the submit mrc-2323 ticket can handle that.

Reply:
It's not really part of mrc-2323, but we should add a new ticket, to bring the finalise page into line with what the add/remove page now does, to emit changes to the workflow metadata, which the wizard manages - and then that's what mrc-2323 will submit. (That data management model is in mrc-2310_reports, which is not yet merged.)

Reply:
New ticket added to the sprint to add those emits of changed metadata to the wizard: https://mrc-ide.myjetbrains.com/youtrack/issue/mrc-2511